### PR TITLE
Portuguese Brazilian (pt-BR) translation

### PR DESCRIPTION
**Pull request by FabsMuller**
Portuguese Brazilian, pt-BR, translation based on v.3.0.1 branch.
The initial efforts started with PR #3100(V.2.0.3) but there were some
problems and the quality needed to be improved. Now I fixed some
failures and reviewed files line by line, double-checked with a text editor
and dictionary updated to the last orthographic agreement of the Portuguese
language; anyway errors still can happen, this is not a robotic translation
and Portuguese has  a Latin branch, so one word in English(sometimes) has
many possible translations in Portuguese; just to complicate a little
bit.

So we are finally making pt-BR available in Node-Red and hope to just
improve from now on.

Resolves: #3100

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by Steve-Mcl** [comment]
Hi @FabsMuller have you signed the EasyCLA as requested in [this comment ↑](https://github.com/node-red/node-red/pull/3804#issuecomment-1194690797)?

**Comment by FabsMuller** [comment]
> Hi @FabsMuller have you signed the EasyCLA as requested in [this comment ↑](https://github.com/node-red/node-red/pull/3804#issuecomment-1194690797)?

Done!

**Comment by Steve-Mcl** [comment]
@herickf, @machadotiago
In the interest of getting pt-BR merged, would you be kind enough to review @FabsMuller contribution please?

Thanks in advance.

File: packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json [hunk 1]
```diff
@@ -0,0 +1,1186 @@
...
+    "languages": {
+        "de": "Alemão",
+        "en-US": "Inglês",
+        "ja": "Japonês",
+        "ko": "Coreano",
+        "pt-BR": "Português(Brasil)",
+        "ru": "Russo",
+        "zh-CN": "Chinês(Simplificado)",
+        "zh-TW": "Chinês(Tradicional)"
+     }
```
Comment (bonanitech) [inline packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json:1185]: There are two sections missing here: `validator` and `contextMenu`

File: packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json [hunk 2]
```diff
@@ -0,0 +1,1186 @@
...
+    },
+    "diagnostics": {
+        "title": "informações do Sistema"
+    },
+    "languages": {
+        "de": "Alemão",
+        "en-US": "Inglês",
+        "ja": "Japonês",
+        "ko": "Coreano",
+        "pt-BR": "Português(Brasil)",
```
Comment (bonanitech) [inline packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json:1181]: I believe this line should be added to the `editor.json` file of other languages.

**Review by bonanitech — COMMENTED** [review]
I found two small issues that I think should be resolved before merging

File: packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json (hunk 1 above)
Comment (FabsMuller) [reply packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json]: Thanks for paying attention to the details Mauricio. Good Job, Already fixed!

File: packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json (hunk 2 above)
Comment (FabsMuller) [reply packages/node_modules/@node-red/editor-client/locales/pt-BR/editor.json:1181]: I did also but at the first moment I think in not touch the other translations but what costs? So I made it, even with Japanese, Russian and Chinese. Obviously that I used a translator to translate "português" to these languages but as it is a single word, not so complicated, I couldn't miss that.

File: packages/node_modules/@node-red/editor-client/locales/ja/editor.json [hunk 3]
```diff
@@ -1179,6 +1179,7 @@
         "en-US": "英語",
         "ja": "日本語",
         "ko": "韓国語",
+        "pt-BR":"ポルトガル語",
```
Comment (kazuhitoyokoi) [inline packages/node_modules/@node-red/editor-client/locales/ja/editor.json:1182]: 👍

Comment (FabsMuller) [reply packages/node_modules/@node-red/editor-client/locales/ja/editor.json:1182]: どうもありがとうございます

**Comment by FabsMuller** [comment]
@Steve-Mcl, when adding "pt-BR" to other languages I notice that some of them are also missing "validator" and "contextMenu" section of **editor.json** in locales too. Probably as they came from 2.x,  If you could please get in touch with the translators to fix that would be nice: Korean - this is missing other sections too, it stops at "editor-tab"  I could even add pt-BR as there was no "languages" section. Russian - No "validator" and "contextMenu". ZH-CN and ZH-TW- No "validator" and "contextMenu" I added "pt-BR" (and "ru") to "languages" section of these ones ("ru" was missing).  If you want  I can talk to them, just gave me their names. Brgds...

**Comment by knolleary** [comment]
@FabsMuller we do not have regular contributors for some of the languages. So yes, some have fallen behind. This is the downside of trying to provide multiple languages in an open source project that relies on contributions from the community.

**Comment by FabsMuller** [comment]
Yes Nick I know about the contributors (following you and Node-red since 0.19), it was not what I meant, in terms of regular, I thought just about the people not as official contribs or anything like that. Sorry if I gave this impression. And I don't consider the translations a downside, it's the opposite. Blessed is a project that has many translations. Means a lot of friends!

File: packages/node_modules/@node-red/nodes/locales/pt-BR/network/21-httprequest.html [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    geralmente não é a coisa certa a se fazer. Se a propriedade <code>msg.headers</code> for deixada inalterada
+    entre os nós, ela será ignorada pelo segundo nó. Para definir cabeçalhos personalizados, <code>msg.headers</code>
+    deve primeiro ser excluído ou redefinido para um objeto vazio: <code>{}</code>.
+    <h4>Tratamento de cookies</h4>
+    <p>A propriedade <code>cookies</code> passada ao nó deve ser um objeto de pares nome/valor.
+    O valor pode ser uma cadeia de caracteres para definir o valor do cookie ou pode ser um
+    objeto com uma única propriedade <code>value</code>.<p>
+    <p>Todos os cookies retornados pela solicitação são devolvidos à propriedade <code>responseCookies</code>.</p>
+    <h4>Tratamento do tipo de conteúdo</h4>
+    <p>Se <code>msg.payload</code> for um objeto, o nó definirá automaticamente o tipo de conteúdo
```
Comment (bonanitech) [inline packages/node_modules/@node-red/nodes/locales/pt-BR/network/21-httprequest.html:79]: It looks like a few paragraphs are missing here

File: packages/node_modules/@node-red/nodes/locales/pt-BR/sequence/17-split.html [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            ordem inversa.</p>  
+         <p>$N é o número de mensagens que chegam - mesmo que sejam idênticas.</p>
+     </dl>
+     <p><b>Exemplo:</b> as configurações a seguir, dada uma sequência de valores numéricos,
+        calcula o valor médio:
+         <ul>
+             <li><b>Reduzir expressão</b>: <code>$A+payload</code></li>
+             <li><b>Valor inicial</b>: <code>0</code></li>
+             <li><b>Expressão de correção</b>: <code>$A/$N</code></li>
+         </ul>
```
Comment (bonanitech) [inline packages/node_modules/@node-red/nodes/locales/pt-BR/sequence/17-split.html:164]: It looks like a section is missing here

**Review by bonanitech — COMMENTED** [review]
I found two other issues. In both cases, there seems to be missing content.

File: packages/node_modules/@node-red/nodes/locales/pt-BR/network/21-httprequest.html (hunk 4 above)
Comment (FabsMuller) [reply packages/node_modules/@node-red/nodes/locales/pt-BR/network/21-httprequest.html:79]: Thanks Again, Mauricio. I double-checked everything, file by file, and found some other small errors that I missed but they are all fixed now.

**Review by caputomarcos — COMMENTED** [review]
Hi  Folks, 

In the languages it would be interesting to include the country.
e.g.

en-US/editor.json
from: `"pt-BR":"Portuguese",`
change to: `"pt-BR":"Portuguese(Brazil)",`

best regards,

Marcos Caputo.

**Comment by FabsMuller** [comment]
> Hi Folks,
> 
> In the languages it would be interesting to include the country. e.g.
> 
> en-US/editor.json from: `"pt-BR":"Portuguese",` change to: `"pt-BR":"Portuguese(Brazil)",`
> 
> best regards,
> 
> Marcos Caputo.

Marcos, I understand your point but as @knolleary is British and uses "en-US : English" himself and not "en-US : English(UK)"; I just follows his standard maybe we could keep as it is and if we get any Portuguese (Portugal) translation; which in my opinion would be unnecessary, as I used a lot of standard (common on both) words, we could change that afterwards.

**Comment by knolleary** [comment]
Thank you @FabsMuller - I appreciate the work you put into this and apologise for the delays. I'm merging to the `dev` branch so it will be in 3.1-beta.2

**Comment by FabsMuller** [comment]
> Thank you @FabsMuller - I appreciate the work you put into this and apologise for the delays. I'm merging to the `dev` branch so it will be in 3.1-beta.2

No problem Nick. I'm here to help. I would like to do more things, but I have to stabilize first. Things were quite hard last year. I'm finding some work into DevOps, so I'm investing more time on this area now, but things will be better this year. I'm quite sure. Have a nice weekend.

**Pull request merged**